sample_apps/rust/examples/utils/query_common.rs

```diff
@@ -0,0 +1,261 @@
+        .await
+        .expect("Failure to get the query client connection with Timestream");
+    tokio::task::spawn(reload.reload_task());
+    Ok(client)
+}
+
+pub fn write(mut file: &fs::File, s: String) -> Result<(), Box<dyn Error>> {
+    let s_formatted = format!("{}\n", s);
+    if file.write(s_formatted.as_bytes()).is_err() {
+        return Err(String::from("Failed to write to file").into());
+    }
+    if file.flush().is_err() {
+        return Err(String::from("Failed to flush file").into());
+    }
+    Ok(())
+}
+
+#[allow(dead_code)]
+pub fn process_scalar_type(data: &types::Datum) -> Result<String, Box<dyn Error>> {
+    data.scalar_value
+        .clone()
+        .ok_or("Scalar value is None".to_string().into())
+}
+
+#[allow(dead_code)]
+pub fn process_time_series_type(
+    data: &[types::TimeSeriesDataPoint],
+    column_info: &types::ColumnInfo,
+) -> Result<String, Box<dyn Error>> {
+    let mut value = String::new();
+    for (i, datum) in data.iter().enumerate() {
+        value.push_str(&datum.time);
+        value.push(':');
+
+        let column_type = column_info.r#type();
+        let column_type_ref = column_type.as_ref().ok_or("Column type is None")?;
+        let scalar_type = column_type_ref.scalar_type.to_owned();
+        let scalar_type_ref = scalar_type.ok_or("Scalar type is None")?;
+
+        let datum_value = datum.value.as_ref().ok_or("Datum value is None")?;
+        let _scalar_value = datum_value
+            .scalar_value
+            .to_owned()
+            .ok_or("Scalar value is None")?;
+
+        if scalar_type_ref.as_str() == "" {
+            value.push_str(&process_scalar_type(datum_value)?);
+        } else if let Some(array_column_info) = &column_type_ref.array_column_info {
+            let array_value = datum_value
+                .array_value
+                .as_ref()
+                .ok_or("Array value is None")?;
+            value.push_str(&process_array_type(array_value, array_column_info)?);
+        } else if let Some(row_column_info) = &column_type_ref.row_column_info {
+            let row_value = datum_value.row_value.as_ref().ok_or("Row value is None")?;
+            value.push_str(&process_row_type(&row_value.data, row_column_info)?);
+        } else {
+            panic!("Bad data type");
+        }
+
+        if i != data.len() - 1 {
+            value.push_str(", ");
+        }
+    }
+    Ok(value)
+}
+
+#[allow(dead_code)]
+pub fn process_array_type(
+    datum_list: &[types::Datum],
+    column_info: &types::ColumnInfo,
+) -> Result<String, Box<dyn Error>> {
+    let mut value = String::new();
+    for (i, datum) in datum_list.iter().enumerate() {
+        let column_type = column_info.r#type();
+        let column_type_ref = column_type.as_ref().ok_or("Column type is None")?;
+
+        let scalar_type = column_type_ref.scalar_type.to_owned();
+        let scalar_type_ref = scalar_type.ok_or("Scalar type is None")?;
+
+        if scalar_type_ref.as_str() != "" {
```

Comment:
If you had clippy in your CI, this would have been caught:
```suggestion
        if !scalar_type_ref.as_str().is_empty() {
```

Reply:
Changed in all places to `is_empty()`.

`bacon clippy --example query_sample` didn't seem to have a problem with this, not sure why.

Reply:
I think it's because of the way we were setting up our modules it was being ignored. I have re-worked so now we have the following structure and bacon clippy works with the submodules with the following command:

`bacon clippy -- --all-targets`

Updated module AST:

```
crate sample_timestream_app
└── mod utils: pub
    ├── mod query_common: pub
    │   ├── struct Args: pub
    │   ├── async fn get_connection: pub
    │   ├── fn process_array_type: pub
    │   ├── fn process_row_type: pub
    │   ├── fn process_scalar_type: pub
    │   ├── fn process_time_series_type: pub
    │   ├── async fn run_query: pub
    │   └── fn write: pub
    └── mod timestream_helper: pub
        ├── struct Args: pub
        ├── async fn create_database: pub
        ├── async fn create_table: pub
        ├── async fn delete_s3_bucket: pub
        └── async fn get_connection: pub
```